### PR TITLE
rp23xx: use small lock in arch/arm/src/rp23xx/rp23xx_usbdev.c

### PR DESCRIPTION
**Pull request by hujun260**
## Summary

rp23xx: use small lock in arch/arm/src/rp23xx/rp23xx_usbdev.c
reason:
We hope to remove all instances of spin_lock_irqsave(NULL).

## Impact
rp23xx

## Testing
ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides a summary of the change, it lacks crucial details.

Here's what's missing:

* **Summary:**  While the "what" is addressed (replacing `spin_lock_irqsave(NULL)`), the "why" and "how" are insufficient.  Why is removing this specific call important?  What alternative locking mechanism is used? How does the new lock impact the code's functionality? What are the benefits of this change (e.g., improved performance, reduced interrupt latency)?

* **Impact:** Simply stating "rp23xx" isn't enough.  Be explicit.  Does this change impact any specific boards or configurations? Are there any user-facing changes?  All the other impact categories (build, hardware, documentation, security, compatibility) are simply omitted.  They should be explicitly stated as NO or YES with explanations if YES.

* **Testing:**  "ci" is not sufficient testing information. While CI testing is important, local testing is also required and should be documented.  Provide details about the local setup used for testing (OS, compiler, target architecture and board).  Crucially, the "Testing logs before change" and "Testing logs after change" sections are empty.  These are essential for demonstrating the effect of the change.  What tests were run? What were the results before and after the change? How do these results confirm the fix?


To make this PR acceptable, you need to provide the missing information in each section. Be specific and provide concrete details to demonstrate that the change is well-motivated, correctly implemented, and thoroughly tested.

File: arch/arm/src/rp23xx/rp23xx_usbdev.c [hunk 1]
```diff
@@ -1713,7 +1714,7 @@ static int rp23xx_epstall_exec(struct usbdev_ep_s *ep)
 
   usbtrace(TRACE_EPSTALL, privep->epphy);
 
-  flags = spin_lock_irqsave(NULL);
+  flags = spin_lock_irqsave(&g_usbdev.lock);
```
Comment (anchao) [inline arch/arm/src/rp23xx/rp23xx_usbdev.c:1717]: Deadlock ...

Comment (hujun260) [reply arch/arm/src/rp23xx/rp23xx_usbdev.c]: done

File: arch/arm/src/rp23xx/rp23xx_usbdev.c [hunk 2]
```diff
@@ -1709,12 +1713,9 @@ static int rp23xx_epcancel(struct usbdev_ep_s *ep,
 static int rp23xx_epstall_exec(struct usbdev_ep_s *ep)
 {
   struct rp23xx_ep_s *privep = (struct rp23xx_ep_s *)ep;
-  irqstate_t flags;
 
   usbtrace(TRACE_EPSTALL, privep->epphy);
 
-  flags = spin_lock_irqsave(NULL);
```
[the remark this answers is not in the document]
Comment (hujun260) [reply arch/arm/src/rp23xx/rp23xx_usbdev.c:1716]: After switching to a smaller lock, it cannot be used recursively, as rp23xx_epstall_exec is already within the lock scope.
8bbca56b-cd42-498e-a151-897c09c3de72

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**